AuthNZ documentation revamp (#62581)
grafana/grafana: docs/sources/administration/planning-iam-strategy/index.md

---
title: Plan your IAM integration strategy
menuTitle: Plan your IAM integration strategy
description: Learn how to plan your identity and access management strategy before setting up Grafana.
weight: 100
keywords:
  - IdP
  - IAM
  - Auth
  - Grafana
---

# Plan your IAM integration strategy

This section describes the decisions you should make when using an Identity and Access Management (IAM) provider to manage access to Grafana. IAM ensures that users have secure access to sensitive data and [other resources]({{< relref "../data-source-management" >}}), simplifying user management and authentication.

## Benefits of integrating with an IAM provider

Integrating with an IAM provider provides the following benefits:

- **User management**: By providing Grafana access to your current user management system, you eliminate the overhead of replicating user information and instead have centralized user management for users' roles and permissions to Grafana resources.

> - **Security**: many IAM solutions provide advanced security features such as multi-factor authentication, RBAC, and audit trails, which can help to improve the security of your Grafana installation.

Reviewer: ```suggestion
- **Security**: Many IAM solutions provide advanced security features such as multi-factor authentication, RBAC, and audit trails, which can help to improve the security of your Grafana installation.
```

> **Note:** Service accounts will eventually replace [API keys]({{< relref ".#api-keys" >}}) as the primary way to authenticate applications that interact with Grafana.

A common use case for creating a service account is to perform operations on automated or triggered tasks. You can use service accounts to:

- Schedule reports for specific dashboards to be delivered on a daily/weekly/monthly basis
- Define alerts in your system to be used in Grafana
- Set up an external SAML authentication provider
- Interact with Grafana without signing in as a user

In [Grafana Enterprise]({{< relref "../../introduction/grafana-enterprise" >}}), you can also use service accounts in combination with [role-based access control]({{< relref "../roles-and-permissions/access-control" >}}) to grant very specific permissions to applications that interact with Grafana.

{{% admonition type="note" %}}
Service accounts can only act in the organization they are created for. We recommend creating service accounts in each organization if you have the same task needed for multiple organizations.
{{% /admonition %}}

> {{< vimeo 742056367 >}}

Reviewer: I recommend adding an intro sentence here like "The following video explains blah blah blah."

To authenticate with Grafana's HTTP API, a randomly generated string known as a service account token can be used as an alternative to a password.

When a service account is created, it can be linked to multiple access tokens. These service access tokens can be utilized in the same manner as API keys, providing a means to programmatically access Grafana HTTP API.

You can create multiple tokens for the same service account. You might want to do this if:

- Multiple applications use the same permissions, but you want to audit or manage their actions separately.
- You need to rotate or replace a compromised token.

{{% admonition type="note" %}}
In Grafana's audit logs it will still show up as the same service account.
{{% /admonition %}}

Service account access tokens inherit permissions from the service account.

### API keys

> **Note:** If you use Grafana v8.5 or newer, you should use service accounts instead of API keys. API keys will be deprecated in the near future. For more information, refer to [Grafana service accounts]({{< relref ".#service-accounts" >}}).

You can use Grafana API keys to interact with data sources via HTTP APIs.

## How to work with roles?

> Grafana utilizes roles to control the access of users and service accounts to specific resources and determine their authorized actions.

Reviewer: ```suggestion
Grafana roles control the access of users and service accounts to specific resources and determine their authorized actions.
```

You can assign roles through the user interface or APIs, establish them through Terraform, or synchronize them automatically via an external IAM provider.

> ### What are roles?

Reviewer: Don't we have role content that you can link to? What you have written is fine, just wondering about adding a link if folks want to explore more?

Within an organization, Grafana has established three primary roles - organization administrator, editor, and viewer - which dictate the user's level of access and permissions, including the ability to edit data sources or create teams.

> To be a member of any organization, every user must be assigned one specific role.

Reviewer: ```suggestion
To be a member of any organization, every user must be assigned a role.
```

> In addition, Grafana offers a server administrator role that grants access to and enables interaction with resources that affect the entire instance, including organizations, users, and server-wide settings.

Reviewer: ```suggestion
In addition, Grafana provides a server administrator role that grants access to and enables interaction with resources that affect the entire instance, including organizations, users, and server-wide settings.
```

This particular role can only be accessed by users of self-hosted Grafana instances. It is a significant role intended for the administrators of the Grafana instance.

> ### What are permissions?

Reviewer: Same here, I think we have permission docs that you could link to. Just as a little extra. Again, what you've provided here is great.

Each role consists of a set of permissions that determine the tasks a user can perform in the system.
For example, the **Admin** role includes permissions that let an administrator create and delete users.

Grafana allows for precise permission settings on both dashboards and folders, giving you the ability to control which users and teams can view, edit, and administer them.
For example, you might want a certain viewer to be able to edit a dashboard. While that user can see all dashboards, you can grant them access to update only one of them.

In [Grafana Enterprise]({{< relref "../../introduction/grafana-enterprise" >}}), you can also grant granular permissions for data sources to control who can query and edit them.

Dashboard, folder, and data source permissions can be set through the UI or APIs or provisioned through Terraform.

> ### Role-based access control

Reviewer: also here, I think we have lots of RBAC docs you can link to.

{{% admonition type="note" %}}
Available in [Grafana Enterprise]({{< relref "../../../introduction/grafana-enterprise" >}}) and [Grafana Cloud](/docs/grafana-cloud/).
{{% /admonition %}}

> If you feel that the basic organization and server administrator roles are too limiting, it may be beneficial to employ RBAC.

Reviewer: ```suggestion
If you think that the basic organization and server administrator roles are too limiting, it might be beneficial to employ role-based access control (RBAC).
```

### Team sync

Team sync is a feature that allows you to synchronize teams or groups from your authentication provider with teams in Grafana. This means that users of specific teams or groups in LDAP, OAuth, or SAML will be automatically added or removed as members of corresponding teams in Grafana. Whenever a user logs in, Grafana will check for any changes in the teams or groups of the authentication provider and update the user's teams in Grafana accordingly. This makes it easy to manage user permissions across multiple systems.

{{% admonition type="note" %}}
Available in [Grafana Enterprise]({{< relref "../../introduction/grafana-enterprise" >}}) and [Grafana Cloud Advanced](/docs/grafana-cloud/).
{{% /admonition %}}

{{% admonition type="note" %}}
Team synchronization occurs only when a user logs in. However, if you are using LDAP, it is possible to enable active background synchronization. This allows for the continuous synchronization of teams.
{{% /admonition %}}

> ### Role Sync

Reviewer: @linoman Are you planning to add content here?

Author: Yes! I forgot this section was missing information.

### Team sync

Team sync is a feature that allows you to synchronize teams or groups from your authentication provider with teams in Grafana. This means that users of specific teams or groups in LDAP, OAuth, or SAML will be automatically added or removed as members of corresponding teams in Grafana. Whenever a user logs in, Grafana will check for any changes in the teams or groups of the authentication provider and update the user's teams in Grafana accordingly. This makes it easy to manage user permissions across multiple systems.

{{% admonition type="note" %}}
Available in [Grafana Enterprise]({{< relref "../../introduction/grafana-enterprise" >}}) and [Grafana Cloud Advanced](/docs/grafana-cloud/).
{{% /admonition %}}

{{% admonition type="note" %}}
Team synchronization occurs only when a user logs in. However, if you are using LDAP, it is possible to enable active background synchronization. This allows for the continuous synchronization of teams.
{{% /admonition %}}

### Role Sync

> Grafana can synchronize basic roles from your authentication provider by mapping attributes from the identity provider to the user role in Grafana. This means that users with specific attributes, like role, team or group membership in LDAP, OAuth, or SAML will be automatically assigned the corresponding role in Grafana. Whenever a user logs in, Grafana will check for any changes in the user information retrieved from the authentication provider and update the user's role in Grafana accordingly.

Reviewer: ```suggestion
Grafana can synchronize basic roles from your authentication provider by mapping attributes from the identity provider to the user role in Grafana. This means that users with specific attributes, like role, team, or group membership in LDAP, OAuth, or SAML, will be automatically assigned the corresponding role in Grafana. Whenever a user logs in, Grafana will check for any changes in the user information retrieved from the authentication provider and update the user's role in Grafana accordingly.
```
